### PR TITLE
migrator view: filter out 'DELETED' invalid records

### PR DESCRIPTION
**Pull request by iulianav**
## Description

Do not return invalid records with collection 'DELETED' to the view.
Also, add a composite index on `valid` and `collection` columns of the
LegacyRecordsMirror table model and create an alembic recipe for it.

## Related Issue

https://its.cern.ch/jira/browse/INSPIR-867
https://its.cern.ch/jira/browse/INSPIR-868

## Motivation and Context

- We are not interested in checking errors for records that have been deleted.
- Indexing on 'collection' and 'valid' will increase the performance of the query.

## Checklist:

- [x] I have all the information that I need (if not, move to `RFC` and look for it).
- [x] I linked the related issue(s) in the corresponding commit logs.
- [x] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [ ] I've added any new docs if API/utils methods were added.
- [ ] I have updated the existing documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Signed-off-by: Iuliana Voinea <iuliana.voinea@student.manchester.ac.uk>

### CODE REVIEW

File: inspirehep/alembic/2f5368ff6d20_add_index_on_legacy_records_mirror_table.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+branch_labels = ()
+depends_on = None
+
+
+def upgrade():
+    """Upgrade database."""
+    op.create_index(
+        'ix_legacy_records_mirror_collection_valid',
+        'legacy_records_mirror',
+        ['collection', 'valid'],
```
Comment (michamos) [inline inspirehep/alembic/2f5368ff6d20_add_index_on_legacy_records_mirror_table.py]: it's probably more useful to have the index in the opposite direction, that is on `('valid', 'collection')`, so that it can be used instead of the current index on `valid` (that you can remove, as it would do double duty with this one).

You can think about an index as something like a phone book (if you know what a phonebook is). That is an index on `(last name, first name)`, and can be used to obtain quickly the phone number of a person if you know either:
* the last name
* the last name and the first name
but it's of no use if you know only the first name.

Here, we want something that can be used to get quickly invalid records, or invalid records and their collection, but we don't need the collection of a record without its validity. That's why the other order is better.

Comment (iulianav) [reply inspirehep/alembic/2f5368ff6d20_add_index_on_legacy_records_mirror_table.py]: okay! Thanks!

File: inspirehep/modules/migrator/models.py [hunk 2]
```diff
@@ -38,6 +38,10 @@
 class LegacyRecordsMirror(db.Model):
     __tablename__ = 'legacy_records_mirror'
 
+    __table_args__ = (
+        db.Index('ix_legacy_records_mirror_valid_collection', 'valid', 'collection'),
+    )
+
     recid = db.Column(db.Integer, primary_key=True, index=True)
     last_updated = db.Column(db.DateTime, default=datetime.utcnow, nullable=False, index=True)
     _marcxml = db.Column('marcxml', db.LargeBinary, nullable=False)
```
Comment (michamos) [inline inspirehep/modules/migrator/models.py]: as I was saying, the whole point of this is that you can now drop the index in `valid` in the next line, because it performs part of the same function as the new index you're adding. You also need to add a migration to remove/add that index on upgrade/downgrade.

**Comment by drjova** [comment]
ci merge please

File: inspirehep/alembic/2f5368ff6d20_update_indices_on_legacy_records_mirror_table.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    op.drop_index(
+        'ix_legacy_records_mirror_valid',
+        'legacy_records_mirror',
+    )
+
+    op.drop_index(
+        'ix_legacy_records_mirror_recid',
+        'legacy_records_mirror',
+    )
```
Comment (michamos) [inline inspirehep/alembic/2f5368ff6d20_update_indices_on_legacy_records_mirror_table.py]: wait, why did you remove this?
